api/api.go

```diff
@@ -347,6 +368,41 @@ func NewClient(config *Config) (*Client, error) {
 		config.HttpClient = defConfig.HttpClient
 	}
```

Comment:
So, with below I can pass my own `HttpClient`, with my own `RoundTripper`

```go
if config.HttpClient == nil {
    config.HttpClient = defConfig.HttpClient
}
```

but the line
```go
config.HttpClient.Transport.(*http.Transport).TLSClientConfig = tlsClientConfig
```

brakes it as it assumes standard `RoundTripper` implementation